api/api_debug_util.go

```diff
@@ -91,3 +116,226 @@ func (api *DebugUtilAPI) PrintBlock(ctx context.Context, blockNrOrHash rpc.Block
 	}
 	return spew.Sdump(block), nil
 }
+
+type CompactionRange struct {
+	Start    hexutil.Bytes `json:"start"`
+	Step     hexutil.Bytes `json:"step"`
+	End      hexutil.Bytes `json:"end"`
+	DB       []string      `json:"db"` // if empty, all database.
+	Interval string        `json:"interval"`
+}
+
+// CompactRange compacts over the given range, for all specified databases.
+func (r *CompactionRange) compactRange(ctx context.Context, chainDB database.DBManager) error {
+	var interval time.Duration
+	if r.Interval != "" {
+		if duration, err := time.ParseDuration(r.Interval); err != nil {
+			return err
+		} else {
+			interval = duration
+		}
+	}
+
+	fn := func(segStart, segEnd []byte) error {
+		logger.Info("Compacting database started", "range", fmt.Sprintf("0x%x-0x%x", segStart, segEnd))
+		cstart := time.Now()
+		if err := r.compactSegment(ctx, chainDB, segStart, segEnd); err != nil {
+			logger.Error("Database compaction failed", "err", err)
+			return err
+		}
+		logger.Info("Compacting database completed", "range", fmt.Sprintf("0x%x-0x%x", segStart, segEnd), "elapsed", common.PrettyDuration(time.Since(cstart)))
```

Comment:
```suggestion
		logger.Info("Compacting database completed", "db", r.DB, "range", fmt.Sprintf("0x%x-0x%x", segStart, segEnd), "elapsed", common.PrettyDuration(time.Since(cstart)))
```